tools/toolchains/gcc.py

```diff
@@ -188,7 +208,10 @@ def assemble(self, source, object, includes):
     @hook_tool
     def compile(self, cc, source, object, includes):
         # Build compile command
-        cmd = cc + self.get_compile_options(self.get_symbols(), includes)
+        if self.check_if_coverage_enabled(source):
+            cmd = cc + self.COVERAGE_FLAGS + self.get_compile_options(self.get_symbols(), includes)
+        else:
+            cmd = cc + self.get_compile_options(self.get_symbols(), includes)
```

Comment:
Nit: Same deal here as I mentioned above, it'd be good to avoid the copy-paste here, especially since this is a sensitive command.